### PR TITLE
Deal with items with same name but different values in ordered field widget

### LINKED ISSUE #75: Not showing 2 options with same name and different values
There is a bug when the vocabulary sends two items with same name and different values.

In our use case we have two content types that when it is translated to Portuguese their names becomes the same.

In this case the widget for some reason swallow one of the two content types instead of showing two content types with the same name.

### PR DESCRIPTION
**Pull request by rodfersou**
fix #75

### CODE REVIEW

**Comment by rodfersou** [comment]
@hvelarde FYI

**Comment by icemac** [comment]
@rodfersou Thank you for your PR. Could you please add a test which assures the changed behaviour?

File: src/z3c/form/browser/orderedselect.py [hunk 1]
```diff
@@ -61,9 +61,9 @@ def deselect(self):
         selecteditems = []
         notselecteditems = []
         for selecteditem in self.selectedItems:
-            selecteditems.append(selecteditem['content'])
+            selecteditems.append(selecteditem['value'])
         for item in self.items:
-            if not item['content'] in selecteditems:
+            if not item['value'] in selecteditems:
```
Comment (hvelarde) [inline src/z3c/form/browser/orderedselect.py]:
```python
if item['value'] not in selecteditems:
```

**Comment by rodfersou** [comment]
@hvelarde done
@icemac can you please point me an widget test example?

There is no test for this widget in this package.

**Comment by icemac** [comment]
@rodfersou The `OrderedSelectWidget` is tested in this doctest: [z3c/form/browser/orderedselect.rst](https://github.com/zopefoundation/z3c.form/blob/master/src/z3c/form/browser/orderedselect.rst). Maybe you could use some of the code in this file to create a new unittest as the doctest also serves as documentation and should not contain too many edge case tests.

**Comment by rodfersou** [comment]
@icemac I am having some trouble to finish this PR https://github.com/zopefoundation/z3c.form/issues/78

**Comment by rodfersou** [comment]
@icemac I think this change in test is sufficiente, please take a look

File: src/z3c/form/browser/orderedselect.rst [hunk 2]
```diff
@@ -109,6 +109,7 @@ providing ``ITerms``. This source uses descriminators wich will fit our setup.
   ...              SimpleVocabulary.createTerm(1, 'a', u'A'),
   ...              SimpleVocabulary.createTerm(2, 'b', u'B'),
   ...              SimpleVocabulary.createTerm(3, 'c', u'C')
```
Comment (icemac) [inline src/z3c/form/browser/orderedselect.rst]: This line is missing a comma, causing the test failures.

**Comment by rodfersou** [comment]
@icemac did you see my changes?

**Comment by icemac** [comment]
@rodfersou Did you already sign the [Zope Committer Agreement](http://docs.zope.org/developer/becoming-a-committer.html)? It will allow you to merge the PR on your own. (That's the policy in the `zopefoundation` repositories.)

**Comment by rodfersou** [comment]
@icemac no, I didn't.. just the Plone Agreement

**Comment by rodfersou** [comment]
@icemac signed and sent the email :-)

**Comment by rodfersou** [comment]
@icemac a release would be appreciated

**Comment by icemac** [comment]
@rodfersou Currently the tests are failing on `master`, see https://travis-ci.org/zopefoundation/z3c.form/jobs/436732823. Are you willing to look into these failures. I think they have to do with new releases of e. g. `zope.schema`.

**Comment by rodfersou** [comment]
@icemac wow! last month they did 6 releases at ``zope.schema``!
can someone guide me on what is going on?

**Comment by jamadden** [comment]
I looked over the test output. Everything looks cosmetic, and all related to raising more specific exceptions with different error messages than before (from both zope.schema and zope.configuration). That's great for code when you can `except SomethingSpecific as ex` and access attributes of `ex` that provide you with actionable information, but it does expose the fragility of doctests like this.

**Comment by rodfersou** [comment]
image

**Comment by davisagli** [comment]
I did some work on fixing the tests for the 3.x branch which can probably be easily merged to master: https://github.com/zopefoundation/z3c.form/commit/4421dc42d1f453f0dc6c976dc0aba4dfba69fc79

**Comment by rodfersou** [comment]
great!
